pkg/jobstore/boltdb/store.go

```diff
@@ -325,7 +335,55 @@ func (b *BoltJobStore) getJobs(tx *bolt.Tx, query jobstore.JobQuery) ([]models.J
 		return nil, err
 	}
 
-	return b.getJobsWithinLimit(result, query), nil
+	// Sort the jobs according to the query.SortBy and query.SortOrder
+	listSorter := func(i, j int) bool {
+		switch query.SortBy {
+		case "id":
+			if query.SortReverse {
+				// what does it mean to sort by ID? (rj) it's mostly meaningless and just
+				// gives us a lexicographic sort of the UUIDs.
+				return result[i].ID > result[j].ID
+			} else {
+				return result[i].ID < result[j].ID
+			}
+		case "created_at":
+			if query.SortReverse {
+				return result[i].CreateTime > result[j].CreateTime
+			} else {
+				return result[i].CreateTime < result[j].CreateTime
+			}
+		default:
+			// We apply modifytime as a default sort so that we can use it for pagination.
+			// Without a known default we won't have a stable sort that makes sense for
+			// offsets/limits.
+			if query.SortReverse {
+				return result[i].ModifyTime > result[j].ModifyTime
+			} else {
+				return result[i].ModifyTime < result[j].ModifyTime
+			}
```

Comment:
I agree `ModifyTime` sounds like a great default for better user experience, but my concern is that it is an always changing value for active jobs and will introduce indexing challenges along with unpredictable user experience.

I think `CreateTime` strikes a good balance between user experience and simplicity. And if that is the case, does it make sense to keep things simple and don't expose `SortBy` option until we have more valid fields to sort by?

Reply:
I think there's still value in sorting by modify time, even if it isn't the default, particularly for just seeing jobs with recent activity.  So maybe just swapping them around for now so that created_at is default and we can expand upon it later?